### DISCUSSION TITLE
Build a bootable non-UEFI ARM64 image (e.g. Raspbian for RPi 4B)?

### DISCUSSION THREAD

**Discussion by septatrix** [systemd/mkosi#2103] [Q&A]
I am aware that an issue might not be the best place to discuss this and a GH discussion would be more appropriate alas those are not enabled on this repo (and the systemd mailing list seems like an even worse place).

I am evaluation the use of mkosi to create images for Raspberry Pi devices. As those do not use UEFI I expect that several things will require manual configuration. Luckily the 64bit version of Raspbian is already using mostly Debian packages so that might simplify things a bit from the view of mkosi (i.e. I do not have to set up a custom distro). For the most part I try to follow the official [`pi-gen` tool](https://github.com/RPi-Distro/pi-gen) as close as possible.

Has anyone ever used mkosi to create such images and has tips how to achieve this goal? (Maybe even with a RPi already.) Any help and pointers would be greatly appreciated - especially regarding the bootloader where I already found https://pete.akeo.ie/2019/07/installing-debian-arm64-on-raspberry-pi.html.

---

@maintainers: I would be happy if you enabled discussions on this repo and transferred this issue.

**Comment by daandemeyer** [comment]
AFAIK nobody from the maintainers has ever tried this. If you don't want UEFI, you'd have to use grub or use Bootable=no and set up everything yourself. That's as far as my knowledge about this goes.

**Comment by adrelanos** [comment]
* https://github.com/grml/grml-debootstrap/issues/114
